cmd/integration/server_test.go

```diff
@@ -0,0 +1,202 @@
+				},
+			},
+		},
+		Status: v1beta2.ResourceReservationStatus{
+			Pods: map[string]string{
+				"driver":     "my-pod-driver",
+				"executor-1": "my-pod-executor-1",
+				"executor-2": "missing-pod-1",
+			},
+		},
+	}
+	driverPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-pod-driver",
+			Namespace: "podNamespace",
+			Labels: map[string]string{
+				common.SparkRoleLabel:  common.Driver,
+				common.SparkAppIDLabel: "appID1",
+			},
+			Annotations: map[string]string{
+				common.DriverCPU:      "1",
+				common.DriverMemory:   "1024Mi",
+				common.ExecutorCPU:    "2",
+				common.ExecutorMemory: "4096Mi",
+				common.ExecutorCount:  "2",
+			},
+		},
+		Spec: v1.PodSpec{
+			NodeName:      "n1",
+			SchedulerName: common.SparkSchedulerName,
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodRunning,
+		},
+	}
+	existingNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "n1",
+			Labels: map[string]string{
+				v1.LabelTopologyZone:          "zone1",
+				v1.LabelFailureDomainBetaZone: "zone1",
+				"resource_channel":            "",
+			},
+		},
+		Status: v1.NodeStatus{
+			Capacity: map[v1.ResourceName]resource.Quantity{
+				"cpu":    resource.MustParse("32"),
+				"memory": resource.MustParse("28192Mi"),
+			},
+			Allocatable: map[v1.ResourceName]resource.Quantity{
+				"cpu":    resource.MustParse("32"),
+				"memory": resource.MustParse("28192Mi"),
+			},
+			Conditions: []v1.NodeCondition{
+				{
+					Type:   v1.NodeReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+	}
+
+	executor1Pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "my-pod-executor-1",
+			Namespace: "podNamespace",
+			Labels: map[string]string{
+				common.SparkRoleLabel:  common.Executor,
+				common.SparkAppIDLabel: "appID1",
+			},
+		},
+		Spec: v1.PodSpec{
+			NodeName: "n1",
+		},
+		Status: v1.PodStatus{
+			Phase: v1.PodRunning,
+		},
+	}
+	allClients := cmd.AllClient{
+		APIExtensionsClient:  apiextensionsfake.NewSimpleClientset(),
+		SparkSchedulerClient: ssclientset.NewSimpleClientset(&rr),
+		KubeClient:           k8sfake.NewSimpleClientset(existingNode, driverPod, executor1Pod),
+	}
+
+	installConfig := config2.Install{
+		Install: config.Install{
+			UseConsoleLog: true,
+		},
+		ShouldScheduleDynamicallyAllocatedExecutorsInSameAZ: true,
+		BinpackAlgo: binpacker.SingleAzMinimalFragmentation,
+	}
+	testSetup := SetUpServer(context.Background(), t, installConfig, allClients)
+	ctx := testSetup.ctx
+	defer testSetup.cleanup()
+	nodeNames := []string{existingNode.Name}
+	args := schedulerapi.ExtenderArgs{
+		Pod: &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "new-podName",
+				Namespace: "podNamespace",
+				Labels: map[string]string{
+					common.SparkRoleLabel:  common.Executor,
+					common.SparkAppIDLabel: "appID1",
+				},
+				Annotations: map[string]string{
+					common.ExecutorCPU:    "2",
+					common.ExecutorMemory: "4096Mi",
+					common.ExecutorCount:  "4",
+				},
```

Comment:
Executors dont have these annotations normally so we should probably remove these here

Reply:
Done